### ISSUE TITLE
Gas multiplier is never applied in truffle-contract

### ISSUE THREAD

**Issue by cgewecke** [trufflesuite/truffle#2539]
## Issue

By default, contracts using autoGas should multiply the gas estimate by 1.25.  At the moment, the multiplying logic seems like it just returns the original estimate value.

Web3 1.2.2 [fixes][1] long-broken logic that registers a transaction as failed when the gas consumed equals the gas sent. In older versions of the eth chain and some contemporary forks ( ETC? maybe Quorum?) failing txs consume all gas and the gas comparison is a reliable indicator that something has gone wrong. 

This is relevant for #2488 because truffle-contract's tests are fueling the tx's with perfect estimates and triggering a false error.  

The question of what behavior is correct is a little complicated and will need further discussion at Web3, but Truffle could side-step this by fixing the multiplier.

Possible that #2512 is related...

cf: [web3 3175](https://github.com/ethereum/web3.js/issues/3175)

## Steps to Reproduce

```javascript
utils.multiplyBigNumberByDecimal(
  utils.bigNumberify(500),
  1.25
);
> 500
```
[1]: https://github.com/ethereum/web3.js/pull/3123/files
## Environment

* Operating System: 
* Ethereum client:
* Truffle version (`truffle version`): 5.0.41
* node version (`node --version`):
* npm version (`npm --version`):

**Comment by eggplantzzz** [comment]
Ahh yes I see what you mean. It looks like the logic in that method isn't sound. Thanks for pointing this out!

**Comment by eggplantzzz** [comment]
A fix for this is now out in the wild with v5.0.44.
